### PR TITLE
Propagate properties in Recording and Sorting OldToNew wrappers

### PR DESCRIPTION
**Pull request by alejoe91**
Implements https://github.com/SpikeInterface/spikeinterface/issues/426

### CODE REVIEW

**Comment by alejoe91** [comment]
@samuelgarcia I can add a test, but we need to add spikeextractors as a dependency. What do you think?

File: spikeinterface/core/old_api_utils.py [hunk 1]
```diff
@@ -44,36 +45,73 @@ def get_channel_property(self, channel_id, property):
...
+                if skip_if_value is not None:
+                    if prop_value == skip_if_value:
+                        del properties[prop_name]
+                        break
+                properties[prop_name]["ids"].append(ch)
+                properties[prop_name]["values"].append(prop_value)
+
+        for property_name, prop_dict in properties.items():
+            property_ids = prop_dict["ids"]
+            property_values = prop_dict["values"]
```
Comment (samuelgarcia) [inline spikeinterface/core/old_api_utils.py:109]: We should also change
`BaseExtractor.set_property(self, key, values, ids=None)`
to
`BaseExtractor.set_property(self, key, values, ids=None, missing=None)`
(missing=None is same behavior as before)
and then here we could handle properly and explicitly missing vales from old API.

File: spikeinterface/core/old_api_utils.py [hunk 2]
```diff
@@ -102,35 +140,56 @@ def get_traces(self, start_frame, end_frame, channel_indices):
...
                              unit_ids=oldapi_sorting_extractor.get_unit_ids())
-               
+
         sorting_segment = OldToNewSortingSegment(oldapi_sorting_extractor)
         self.add_sorting_segment(sorting_segment)
-        
+
         self.is_dumpable = False
+
+        # add old properties
```
Comment (samuelgarcia) [inline spikeinterface/core/old_api_utils.py:169]: This looks like a sad copy/paste.
You should make a function for the 2 situtations.

**Comment by samuelgarcia** [comment]
Not super happy to add spikeextractors for tests suite as new dep.
It will be tested anyway in nwb_conversion tools.

File: spikeinterface/core/old_api_utils.py (hunk 1 above)
Comment (alejoe91) [reply spikeinterface/core/old_api_utils.py:109]: Let's make a separate PR for that. I don't understand exactly what you mean

File: spikeinterface/core/old_api_utils.py [hunk 3]
```diff
@@ -139,19 +158,61 @@ class OldToNewSortingSegment(BaseSortingSegment):
...
+                prop_name = old_to_new_property_map[prop]["name"]
+                skip_if_value = old_to_new_property_map[prop]["skip_if_value"]
+            else:
+                prop_name = prop
+
+            if prop_name not in properties:
+                properties[prop_name] = dict()
+                properties[prop_name]["ids"] = []
+                properties[prop_name]["values"] = []
+            if skip_if_value is not None:
```
[the remark this answers is not in the document]
Comment (alejoe91) [reply spikeinterface/core/old_api_utils.py]: done!

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**